drift_dev/lib/src/writer/tables/table_writer.dart

```diff
@@ -50,9 +50,26 @@ abstract class TableOrViewWriter {
     if (column.customConstraints != null) {
       additionalParams['\$customConstraints'] =
           asDartLiteral(column.customConstraints!);
-    } else if (constraints.isNotEmpty) {
+    } else if (constraints.values.any((constraint) => constraint.isNotEmpty)) {
       // Use the default constraints supported by drift
-      additionalParams['defaultConstraints'] = asDartLiteral(constraints);
+
+      final literalEntries = [
+        for (final entry in constraints.entries)
+          'SqlDialect.${entry.key.name}: ${asDartLiteral(entry.value)},',
+      ];
+
+      additionalParams['defaultConstraints'] = '''(context) {
```

Comment:
This generates a lot of code. I think we should look for ways to optimize common cases here:

1. If the constraints are the same for all dialects: Maybe there could be a method like
```dart
// Maybe a static method in GeneratedColumn
void Function(GenerationContext) always(String sql) {
  return (context) => context.write(sql);
}
// we can now do something like `defaultConstraints: GeneratedColumn.always('PRIMARY KEY AUTOINCREMENT')`, saving a lot of code
```
2. There could be a similar helper method taking a `Map<SqlDialect, String>`. Essentially we'd be turning the current code snippet into a single method which is then called in the generated code, avoiding a lot of duplication.